AWS CC bindings can't be compiled

<!--- Please keep this note for the community --->

### Community Note

- Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
- Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
- If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### cdktf & Language Versions

<!--- Please make note of the version you are using for the `hashicorp/terraform-cdk` and language bindings you are using.
If you are not running the latest version, please upgrade because your issue may have already been fixed. --->

### Affected Resource(s)

<!--- Please list the affected resources. --->

### Debug Output

<!---
Please provide a link to a GitHub Gist containing the complete debug output. Please do NOT paste the debug output in the issue; just paste a link to the Gist.
--->

### Expected Behavior

<!--- What should have happened? --->

Should compile

### Actual Behavior

<!--- What actually happened? --->

```
OTE: Temp directory retained due to an error: /var/folders/y5/nm6gc8rj0p76ds9pwyfz2xym0000gn/T/temp-numPox
Error: jsii compilation failed with non-zero exit code: 1
  | [2023-01-06T13:38:34.682] [ERROR] jsii/compiler - Compilation errors prevented the JSII assembly from being created
  | warning JSII6: A "peerDependency" on "constructs" at "10.1.183" means you should take a "devDependency" on "constructs" at "10.1.183" (found "undefined")
  | warning JSII6: A "peerDependency" on "cdktf" at "0.14.3" means you should take a "devDependency" on "cdktf" at "0.14.3" (found "undefined")
  | warning JSII3: There is no "README.md" file. It is required in order to generate valid PyPI (Python) packages.
  | providers/awscc/backup-backup-plan/index.ts:94:5 - error TS2740: Type '{ [key: string]: string; }' is missing the following properties from type 'StringMap': terraformResource, terraformAttribute, lookup, computeFqn, and 3 more.
  | 94     return this.getStringMapAttribute('backup_options');
  |        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  | providers/awscc/databrew-job/index.ts:2089:5 - error TS2322: Type '{ [key: string]: string; }' is not assignable to type 'StringMap'.
  | 2089     return this.getStringMapAttribute('parameters');
  |          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  | providers/awscc/databrew-job/index.ts:2420:5 - error TS2322: Type '{ [key: string]: string; }' is not assignable to type 'StringMap'.
  | 2420     return this.getStringMapAttribute('parameters');
  |          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  | providers/awscc/iot-topic-rule/index.ts:2853:5 - error TS2322: Type '{ [key: string]: string; }' is not assignable to type 'StringMap'.
  | 2853     return this.getStringMapAttribute('client_properties');
  |          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  | providers/awscc/iot-topic-rule/index.ts:7680:5 - error TS2322: Type '{ [key: string]: string; }' is not assignable to type 'StringMap'.
  | 7680     return this.getStringMapAttribute('client_properties');
  |          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  +----------------------------------------------------------------------------------
  | Command: /usr/local/Cellar/cdktf/0.14.3/libexec/lib/node_modules/cdktf-cli/node_modules/jsii/bin/jsii --silence-warnings reserved-word
  | Workdir: /var/folders/y5/nm6gc8rj0p76ds9pwyfz2xym0000gn/T/temp-numPox
  +----------------------------------------------------------------------------------
    at p (/usr/local/Cellar/cdktf/0.14.3/libexec/lib/node_modules/cdktf-cli/bundle/bin/cmds/handlers.js:51:1581)
    at ChildProcess.<anonymous> (/usr/local/Cellar/cdktf/0.14.3/libexec/lib/node_modules/cdktf-cli/bundle/bin/cmds/handlers.js:54:112)
    at Object.onceWrapper (node:events:628:26)
    at ChildProcess.emit (node:events:513:28)
    at ChildProcess.emit (node:domain:489:12)
    at ChildProcess._handle.onexit (node:internal/child_process:291:12)
⠋ downloading and generating modules and providers...
jsii compilation failed with non-zero exit code: 1
  | [2023-01-06T13:38:34.682] [ERROR] jsii/compiler - Compilation errors prevented the JSII assembly from being created
  | warning JSII6: A "peerDependency" on "constructs" at "10.1.183" means you should take a "devDependency" on "constructs" at "10.1.183" (found "undefined")
  | warning JSII6: A "peerDependency" on "cdktf" at "0.14.3" means you should take a "devDependency" on "cdktf" at "0.14.3" (found "undefined")
  | warning JSII3: There is no "README.md" file. It is required in order to generate valid PyPI (Python) packages.
  | providers/awscc/backup-backup-plan/index.ts:94:5 - error TS2740: Type '{ [key: string]: string; }' is missing the following properties from type 'StringMap': terraformResource, terraformAttribute, lookup, computeFqn, and 3 more.
  | 94     return this.getStringMapAttribute('backup_options');
  |        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  | providers/awscc/databrew-job/index.ts:2089:5 - error TS2322: Type '{ [key: string]: string; }' is not assignable to type 'StringMap'.
  | 2089     return this.getStringMapAttribute('parameters');
  |          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  | providers/awscc/databrew-job/index.ts:2420:5 - error TS2322: Type '{ [key: string]: string; }' is not assignable to type 'StringMap'.
  | 2420     return this.getStringMapAttribute('parameters');
  |          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  | providers/awscc/iot-topic-rule/index.ts:2853:5 - error TS2322: Type '{ [key: string]: string; }' is not assignable to type 'StringMap'.
  | 2853     return this.getStringMapAttribute('client_properties');
  |          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  | providers/awscc/iot-topic-rule/index.ts:7680:5 - error TS2322: Type '{ [key: string]: string; }' is not assignable to type 'StringMap'.
  | 7680     return this.getStringMapAttribute('client_properties');
  |          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  +----------------------------------------------------------------------------------
  | Command: /usr/local/Cellar/cdktf/0.14.3/libexec/lib/node_modules/cdktf-cli/node_modules/jsii/bin/jsii --silence-warnings reserved-word
  | Workdir: /var/folders/y5/nm6gc8rj0p76ds9pwyfz2xym0000gn/T/temp-numPox
  +----------------------------------
``` 

### Steps to Reproduce

<!--- Please list the steps required to reproduce the issue. --->

Use this `cdktf.json` file:


```json
{
  "language": "python",
  "app": "pipenv run python main.py",
  "projectId": "2d650884-83c6-444f-9726-094c4a98476f",
  "sendCrashReports": "false",
  "terraformProviders": [
    {
      "name": "awscc",
      "source": "hashicorp/awscc",
      "version": "0.33.0"
    }
  ],
  "terraformModules": [],
  "codeMakerOutput": "imports",
  "context": {
    "excludeStackIdFromLogicalIds": "true",
    "allowSepCharsInLogicalIds": "true"
  }
}
```

Run `cdktf get`

### Important Factoids

<!--- Are there anything atypical about your accounts that we should know? --->

### References

<!---
Information about referencing Github Issues: https://help.github.com/articles/basic-writing-and-formatting-syntax/#referencing-issues-and-pull-requests

Are there any other GitHub issues (open or closed) or pull requests that should be linked here? Vendor documentation? For example:
--->

- [Discuss](https://discuss.hashicorp.com/t/bug-in-awscc-upgrade-from-0-32-0-to-0-33-0-and-onwards/48648)

--- a/packages/@cdktf/provider-generator/lib/get/generator/emitter/attributes-emitter.ts
+++ b/packages/@cdktf/provider-generator/lib/get/generator/emitter/attributes-emitter.ts
@@ -167,6 +167,25 @@ export class AttributesEmitter {
     if (type.isBoolean) {
       return `this.getBooleanAttribute('${att.terraformName}')`;
     }
+
+    // TODO: enable this change once we want to release the next major version
+    // return `this.getComputed${uppercaseFirst(att.mapType)}MapAttribute('${
+    //   att.terraformName
+    // }')`;
+    // if (type.isComputedStringMap) return `this.getComputedStringMapAttribute('${att.terraformName}')`;
+    // if (type.isComputedNumberMap) return `this.getComputedNumberMapAttribute('${att.terraformName}')`;
+    // if (type.isComputedBooleanMap) return `this.getComputedBooleanMapAttribute('${att.terraformName}')`;
+    // if (type.isComputedAnyMap) return `this.getComputedAnyMapAttribute('${att.terraformName}')`;
+
+    if (type.isComputedStringMap)
+      return `new cdktf.StringMap(this, '${att.terraformName}')`;
+    if (type.isComputedNumberMap)
+      return `new cdktf.NumberMap(this, '${att.terraformName}')`;
+    if (type.isComputedBooleanMap)
+      return `new cdktf.BooleanMap(this, '${att.terraformName}')`;
+    if (type.isComputedAnyMap)
+      return `new cdktf.AnyMap(this, '${att.terraformName}')`;
+
     if (type.isMap) {
       return `this.get${uppercaseFirst(att.mapType)}MapAttribute('${
         att.terraformName

--- a/packages/@cdktf/provider-generator/lib/get/generator/emitter/attributes-emitter.ts
+++ b/packages/@cdktf/provider-generator/lib/get/generator/emitter/attributes-emitter.ts
@@ -169,6 +169,7 @@ export class AttributesEmitter {
     }
 
     // TODO: enable this change once we want to release the next major version
+    // https://github.com/hashicorp/terraform-cdk/issues/2532
     // return `this.getComputed${uppercaseFirst(att.mapType)}MapAttribute('${
     //   att.terraformName
     // }')`;

--- a/packages/cdktf/lib/terraform-resource.ts
+++ b/packages/cdktf/lib/terraform-resource.ts
@@ -155,6 +155,24 @@ export class TerraformResource
     return Token.asAnyMap(this.interpolationForAttribute(terraformAttribute));
   }
 
+  // TODO: enable this change once we want to release the next major version
+  // Changing APIs like this is a breaking change in python
+  // public getComputedStringMapAttribute(terraformAttribute: string) {
+  //   return new StringMap(this, terraformAttribute);
+  // }
+
+  // public getComputedNumberMapAttribute(terraformAttribute: string) {
+  //   return new NumberMap(this, terraformAttribute);
+  // }
+
+  // public getComputedBooleanMapAttribute(terraformAttribute: string) {
+  //   return new BooleanMap(this, terraformAttribute);
+  // }
+
+  // public getComputedAnyMapAttribute(terraformAttribute: string) {
+  //   return new AnyMap(this, terraformAttribute);
+  // }
+
   public get terraformMetaArguments(): { [name: string]: any } {
     assert(
       !this.forEach || typeof this.count === "undefined",

--- a/packages/cdktf/lib/terraform-resource.ts
+++ b/packages/cdktf/lib/terraform-resource.ts
@@ -156,6 +156,7 @@ export class TerraformResource
   }
 
   // TODO: enable this change once we want to release the next major version
+  // https://github.com/hashicorp/terraform-cdk/issues/2532
   // Changing APIs like this is a breaking change in python
   // public getComputedStringMapAttribute(terraformAttribute: string) {
   //   return new StringMap(this, terraformAttribute);

--- a/test/provider-tests/providers.json
+++ b/test/provider-tests/providers.json
@@ -8,7 +8,7 @@
     "nomad": "nomad",
     "external": "external",
     "openstack": "terraform-providers/openstack",
-    "oci": "hashicorp/oci@= 4.75.0",
-    "datadog": "DataDog/datadog@~> 3.0",
-    "awscc": "awscc@= 0.17.0"
+    "oci": "hashicorp/oci",
+    "datadog": "DataDog/datadog",
+    "awscc": "awscc"
 }

